### PR TITLE
marp-cli: init at 3.1.0

### PR DESCRIPTION
**Pull request by GuillaumeDesforges**
###### Description of changes

Add the latest release of Marp, a tool to transform markdown into slideshows (HTML, PDF, ...)

https://marp.app/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by GuillaumeDesforges** [comment]
ofborg-eval fails but I thought it had been solved in https://github.com/NixOS/nixpkgs/pull/214952 😕 

cc @lilyinstarlight

**Comment by lilyinstarlight** [comment]
> ofborg-eval fails but I thought it had been solved in #214952 😕

This is a different issue, unfortunately. You still need to manually fetch deps with `offlineCache = fetchYarnDeps { ... }` and also vendor `package.json` with yarn2nix, but I was cooking up a patch to allow it to work without vendoring `package.json` or IFD and updating the documentation to describe such

But for now, I guess you'll just have to vendor it

File: pkgs/tools/typesetting/marp/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "marp-team";
+    repo = "marp-cli";
+    rev = "v${version}";
+    sha256 = "2TsiHv598uxbns8CAz5mljSsFU3gFuHaZpBaaE2KMkE=";
+  };
+
+  packageJSON = ./package.json;
+  yarnLock = ./yarn.lock;
+  yarnNix = ./yarn.nix;
```
Comment (lilyinstarlight) [inline pkgs/tools/typesetting/marp/default.nix:21]: You can actually get by without `yarn.lock` or `yarn.nix` (which are large-ish files) via something like this:

```suggestion
  offlineCache = fetchYarnDeps {
    yarnLock = "${src}/yarn.lock";
    hash = "sha256-AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA=";
  };
```

Comment (GuillaumeDesforges) [reply pkgs/tools/typesetting/marp/default.nix:21]: Replaced with 

```diff
--- a/pkgs/tools/typesetting/marp/default.nix
+++ b/pkgs/tools/typesetting/marp/default.nix
@@ -1,14 +1,13 @@
 {
   mkYarnPackage
+, fetchYarnDeps
 , fetchFromGitHub
 , lib
 }:

-let version = "3.1.0";
-in
-mkYarnPackage {
+mkYarnPackage rec {
   name = "marp-cli";
-  version = version;
+  version = "3.1.0";
   src = fetchFromGitHub {
     owner = "marp-team";
     repo = "marp-cli";
@@ -17,8 +16,10 @@ mkYarnPackage {
   };

   packageJSON = ./package.json;
-  yarnLock = ./yarn.lock;
-  yarnNix = ./yarn.nix;
+  offlineCache = fetchYarnDeps {
+    yarnLock = "${src}/yarn.lock";
+    hash = "sha256-TJ8fpnsbcSG2gYvQyCMWKMyMxaybVi3LB5Ea2F1elDw=";
+  };
```

got 

```
error: builder for '/nix/store/k0rmfjfr8km9drpikvmz3dj8dh5vxh3w-marp-team-marp-cli-modules-3.1.0.drv' failed with exit code 1;
       last 10 log lines:
       >     at Generator.next (<anonymous>)
       >     at step (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:310:30)
       >     at /nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:321:13
       > Error: Couldn't find any versions for "rollup" that matches "^3.26.3" in our cache (possible versions are ""). This is usually caused by a missing entry in the lockfile, running Yarn without the --offline flag may help fix this issue.
       >     at MessageError.ExtendableBuiltin (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:721:66)
       >     at new MessageError (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:750:123)
       >     at NpmResolver.<anonymous> (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:50438:15)
       >     at Generator.next (<anonymous>)
       >     at step (/nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:310:30)
       >     at /nix/store/qan7j0r0jx44ax6i4ps9fnxdwjv5k9b4-yarn-1.22.19/libexec/yarn/lib/cli.js:321:13
       For full logs, run 'nix log /nix/store/k0rmfjfr8km9drpikvmz3dj8dh5vxh3w-marp-team-marp-cli-modules-3.1.0.drv'.
error: 1 dependencies of derivation '/nix/store/g6v5g8pclsvmm11ajpnjwhp0cc5vhs1h-marp-cli.drv' failed to build
```

**Comment by GuillaumeDesforges** [comment]
@lilyinstarlight given the [current issue with `offlineCache`](https://github.com/NixOS/nixpkgs/pull/244504#discussion_r1275102198), could we get this first iteration merged?

**Comment by GuillaumeDesforges** [comment]
Closing in favor of https://github.com/NixOS/nixpkgs/pull/248107

**Pull request closed without merging**